src/consensus/aft/raft.h

```diff
@@ -883,9 +872,27 @@ namespace aft
           }
         }
 
+        size_t backup_ack_timeout_count = 0;
         for (auto& node : nodes)
         {
           node.second.last_ack_timeout += elapsed;
+          if (node.second.last_ack_timeout >= election_timeout)
+          {
+            backup_ack_timeout_count++;
+          }
+        }
+
+        if (backup_ack_timeout_count >= get_quorum(nodes.size()))
+        {
+          // CheckQuorum: The primary automatically steps down if it has not
+          // heard back from a majority of backups during an election timeout.
+          LOG_INFO_FMT(
+            "Stepping down as follower {}: No ack received from a majority {} "
+            "of backups in last {}",
+            state->my_node_id,
+            backup_ack_timeout_count,
+            election_timeout);
+          become_follower();
```

Comment:
Note that this calls `become_follower()` rather than `become_aware_of_new_term(term)`. This is because this is a weaker transition than finding out about a new term. In particular, the stepping down node still keeps track of who it voted for in this term.